ps_imageslider.php

```diff
@@ -70,57 +75,130 @@ public function __construct()
-                $res &= Configuration::updateValue('HOMESLIDER_WRAP', $this->default_wrap, false, $shop_group_id, $shop_id);
+            if (!in_array($shop_group_id, $shop_groups_list)) {
+                $shop_groups_list[] = $shop_group_id;
             }
 
-            /* Sets up Shop Group configuration */
-            if (count($shop_groups_list)) {
-                foreach ($shop_groups_list as $shop_group_id) {
-                    $res &= Configuration::updateValue('HOMESLIDER_SPEED', $this->default_speed, false, $shop_group_id);
-                    $res &= Configuration::updateValue('HOMESLIDER_PAUSE_ON_HOVER', $this->default_pause_on_hover, false, $shop_group_id);
-                    $res &= Configuration::updateValue('HOMESLIDER_WRAP', $this->default_wrap, false, $shop_group_id);
-                }
+            /* Sets up configuration */
+            $res = Configuration::updateValue('HOMESLIDER_SPEED', $this->default_speed, false, $shop_group_id, $shop_id);
+            $res &= Configuration::updateValue('HOMESLIDER_PAUSE_ON_HOVER', $this->default_pause_on_hover, false, $shop_group_id, $shop_id);
+            $res &= Configuration::updateValue('HOMESLIDER_WRAP', $this->default_wrap, false, $shop_group_id, $shop_id);
+        }
+
+        /* Sets up Shop Group configuration */
+        if (count($shop_groups_list)) {
+            foreach ($shop_groups_list as $shop_group_id) {
+                $res &= Configuration::updateValue('HOMESLIDER_SPEED', $this->default_speed, false, $shop_group_id);
+                $res &= Configuration::updateValue('HOMESLIDER_PAUSE_ON_HOVER', $this->default_pause_on_hover, false, $shop_group_id);
+                $res &= Configuration::updateValue('HOMESLIDER_WRAP', $this->default_wrap, false, $shop_group_id);
             }
+        }
+
+        /* Sets up Global configuration */
+        $res &= Configuration::updateValue('HOMESLIDER_SPEED', $this->default_speed);
+        $res &= Configuration::updateValue('HOMESLIDER_PAUSE_ON_HOVER', $this->default_pause_on_hover);
+        $res &= Configuration::updateValue('HOMESLIDER_WRAP', $this->default_wrap);
 
-            /* Sets up Global configuration */
-            $res &= Configuration::updateValue('HOMESLIDER_SPEED', $this->default_speed);
-            $res &= Configuration::updateValue('HOMESLIDER_PAUSE_ON_HOVER', $this->default_pause_on_hover);
-            $res &= Configuration::updateValue('HOMESLIDER_WRAP', $this->default_wrap);
+        /* Creates tables */
+        $res &= $this->createTables();
+
+        /* Adds samples */
+        if ($res) {
+            $this->installSamples();
+        }
 
-            /* Creates tables */
-            $res &= $this->createTables();
+        // Disable on mobiles and tablets
+        $this->disableDevice(Context::DEVICE_MOBILE);
 
-            /* Adds samples */
-            if ($res) {
-                $this->installSamples();
+        return (bool)$res;
```

Comment:
$res is already a boolean because of `$res &=` :thinking:

Reply:
Updated. That was coming from the previous version, that's why I did not change it.